### ISSUE TITLE
Varnish 4.1.2 - Body not included on bereq after std.CacheReqBody()

### ISSUE THREAD

**Issue by EliezerIsrael** [aondio/libvmod-bodyaccess#10]
I'm trying to use libvmod-bodyaccess to cache post requests. 

I'm finding that the call to std.CacheReqBody() consumes the request body, and it doesn't get passed on to the back-end request.  I don't see any indication of how to populate the bereq with a body. 

I understood the use case of caching post requests to be exactly the purpose of this module, but I don't seem to be able to construct a complete system.  How can I both cache the request body, so as to use it in the hash, and still pass it through on the back end?

**Comment by EliezerIsrael** [comment]
vcl

```
vcl 4.0;
import std;
import bodyaccess;

backend default {
    .host = "127.0.0.1";
    .port = "8080";
    .first_byte_timeout = 900s;
}
backend node {
    .host = "127.0.0.1";
    .port = "4040";
    .first_byte_timeout = 900s;
}

sub vcl_recv {
    if (std.port(server.ip) == 4039) {
        set req.backend_hint = node;
        if (req.method == "POST") {
           set req.http.x-method = req.method;
           std.cache_req_body(100KB);
        }
        return(hash);
    }
}

sub vcl_hash {
    if (req.method == "POST") {
    bodyaccess.hash_req_body();
    }
}

sub vcl_backend_fetch {
    set bereq.method = bereq.http.x-method;
}
```

POST Requests get hashed and cached, but the content size on the req is way bigger than content size of bereq, and the body is blank on the backend.

**Comment by aondio** [comment]
Hi, 
in 4.1 this bug: https://github.com/varnishcache/varnish-cache/issues/1927 is currently open.
Once it will be fixed it all be possible to cache and pass the body to the backend.

**Comment by fgsch** [comment]
Now that #1927 is fixed we should be able to do this.

**Comment by aondio** [comment]
Hi @EliezerIsrael, this has been fixed on the 5.0 branch.
